src/screens/ShipItemDetails/index.tsx

```diff
@@ -145,53 +153,38 @@ const ShipItemDetails = () => {
           </View>
         </View>
       </View>
-      <Text>{'Container'}</Text>
-      <AutoInputInternalLocation
-        label="AutoInputInternalContainer"
-        data={state.containerList ? state.containerList : []}
-        selectedContainerItem={selectedContainerItem}
-        selectedData={(selectedItem: any, index: number) => {
-          setSelectedContainerItem(index);
-          state.containerId = selectedItem;
-          const container = state.shipmentDetails.availableContainers[index];
-          setState({
-            ...state,
-            containerId: selectedItem,
-            container: container,
-            selectedContainerItem: index,
-          });
-        }}
-      />
+      <View style={styles.containerField}>
+        <Text>{'Container'}</Text>
+        <AutoInputInternalLocation
+          label="AutoInputInternalContainer"
+          data={state.containerList ? state.containerList : []}
+          selectedContainerItem={selectedContainerItem}
+          selectedData={(selectedItem: any, index: number) => {
+            setSelectedContainerItem(index);
+            state.containerId = selectedItem;
+            const container = state.shipmentDetails.availableContainers[index];
```

Comment:
What's the point of looking for the container by index. You already have the `selectedItem` here. Please refactor it. Remove `state.container` and `state.containerId`. Store properly `selectedItem` into `state.selectedContainerItem` instead index. And use it in the request.